File: 1-Iniciante/2709-As-Moedas-de-Robbie/2709.cpp
#include <iostream>
#include <cmath>

bool isprime(int n) {
    if (n < 2)
        return false;
    for (int i = 2; i <= sqrt(n); ++i) {
        if (n % i == 0)
            return false;
    }
    return true;
}

int main() {
    int len;
    while (std::cin>>len) {
        int coin, increment, sum = 0, coins[len];
        for (int i = 0; i < len; i++) {
            std::cin>>coin;
            coins[i] = coin;
        }
        std::cin>>increment;
        for (int i = len-1; i >= 0; i -= increment) {
            sum += coins[i];
        }
        if (isprime(sum))
            std::cout<<"You’re a coastal aircraft, Robbie, a large silver aircraft.\n";
        else
            std::cout<<"Bad boy! I’ll hit you.\n";   
    }
    return 0;
}

File: 1-Iniciante/2709-As-Moedas-de-Robbie/2709.java
import java.util.Scanner;
import java.lang.Math;

public class Main {
    public static boolean prime(int num) {
        if (num < 2) {
            return false;
        }
        for (int i = 2; i <= Math.sqrt(num); ++i) {
            if (num % i == 0) {
                return false;
            }
        }
        return true;
    }
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        int coins, jump;
        while (scan.hasNext()) {
            coins = scan.nextInt();
            int values[] = new int[coins+1], sum = 0;
            for (int i = 0; i < coins; ++i) {
                values[i] = scan.nextInt();
            }
            jump = scan.nextInt();
            for (int i = coins - 1; i >= 0; i -= jump) {
                sum += values[i];
            }
            if (prime(sum)) {
                System.out.println("You’re a coastal aircraft, Robbie, a large silver aircraft.");
            }
            else {
                System.out.println("Bad boy! I’ll hit you.");
            }
        }
    }
}